### PR TITLE
Added Web5 Connect Server

### PR DESCRIPTION
**Pull request by thehenrytsai**
1. Ported web5 connect server into this repo.
1. Separated business logic from `express` into a `Web5ConnectServer` class.
1. Added tests for 100% new code coverage.
1. Added issue to track TODOs.

### CODE REVIEW

**Review by LiranCohen — COMMENTED** [review]
Nice! One step closer!!

Just some nits about using `loglevel` since that's used in most places for logging.

Another thing I noticed, since fetching the request/response is stateful should we:
a) have google create sticky sessions for these requests?
b) use Redis as the cache mechanism?

Also, haven't looked too much into it since it's used sparsely on this repo, are you familiar with the request counter that exists? Do we want to count these requests/errors as well? I feel like logging should be enough tbh.

File: src/http-api.ts [hunk 1]
```diff
@@ -361,6 +370,86 @@ export class HttpApi {
...
+      console.log('Storing Pushed Authorization Request (PAR) request...');
+
+      const result = await this.web5ConnectServer.setWeb5ConnectRequest(req.body.request);
+      res.status(201).json(result);
+    });
+
+    /**
+     * Endpoint that the Identity Provider (wallet) calls to retrieve the Pushed Authorization Request.
+     */
+    this.#api.get('/connect/:requestId.jwt', async (req, res) => {
```
[the remark this answers is not in the document]
Comment (thehenrytsai) [reply src/http-api.ts:387]: This route came from Frank. = ]

File: src/http-api.ts [hunk 2]
```diff
@@ -361,6 +370,86 @@ export class HttpApi {
     }
   }
 
+  #setupWeb5ConnectServerRoutes(): void {
+    /**
+     * Endpoint that the connecting App pushes the Pushed Authorization Request Object to start the Web5 Connect flow.
+     */
+    this.#api.post('/connect/par', async (req, res) => {
+      console.log('Storing Pushed Authorization Request (PAR) request...');
```
[the remark this answers is not in the document]
Comment (thehenrytsai) [reply src/http-api.ts:378]: Done.

File: src/http-api.ts [hunk 3]
```diff
@@ -361,6 +370,86 @@ export class HttpApi {
...
+
+      const result = await this.web5ConnectServer.setWeb5ConnectRequest(req.body.request);
+      res.status(201).json(result);
+    });
+
+    /**
+     * Endpoint that the Identity Provider (wallet) calls to retrieve the Pushed Authorization Request.
+     */
+    this.#api.get('/connect/:requestId.jwt', async (req, res) => {
+      console.log(`Retrieving Web5 Connect Request object of ID: ${req.params.requestId}...`);
```
[the remark this answers is not in the document]
Comment (thehenrytsai) [reply src/http-api.ts:388]: Done.

File: src/http-api.ts [hunk 4]
```diff
@@ -361,6 +370,86 @@ export class HttpApi {
...
+        res.set('Content-Type', 'application/jwt');
+        res.send(requestObjectJwt);
+      }
+    });
+
+    /**
+     * Endpoint that the Identity Provider (wallet) pushes the Authorization Response ID token to.
+     */
+    this.#api.post('/connect/sessions', async (req, res) => {
+      console.log('Storing Identity Provider (wallet) pushed response with ID token...');
```
[the remark this answers is not in the document]
Comment (thehenrytsai) [reply src/http-api.ts:408]: Done.

File: src/http-api.ts [hunk 5]
```diff
@@ -361,6 +370,86 @@ export class HttpApi {
...
+        });
+      }
+    });
+
+    /**
+     * Endpoint that the connecting App polls to check if the Identity Provider (Wallet) has posted the Web5 Connect Response object.
+     * The Web5 Connect Response is also an ID token.
+     */
+    this.#api.get('/connect/sessions/:state.jwt', async (req, res) => {
+      console.log(`Retrieving ID token for state: ${req.params.state}...`);
```
[the remark this answers is not in the document]
Comment (thehenrytsai) [reply src/http-api.ts:436]: Done.

**Review by LiranCohen — APPROVED** [review]
🎸 🐐 🚀

**Pull request merged**